msteams-platform/bots/how-to/conversations/conversation-messages.md

```diff
@@ -215,6 +223,312 @@ async def on_members_added_activity(
+
+    this.updateCurrentMessage(messageId, undeletedMessage);
+}
+
+```
+
+---
+
+## Soft delete message
+
+When you soft delete a message in a chat, the bot gets a notification of the soft delete message event.
+
+To get a soft delete message event notification in a bot, you can override the `OnMessageDeleteActivityAsync` handler.
+
+The following is an example of a soft delete message event notification when a message is deleted:
+
+# [C#](#tab/csharp5)
+
+```csharp
+
+protected override async Task OnTeamsMessageSoftDeleteAsync(ITurnContext<IMessageDeleteActivity> turnContext, CancellationToken cancellationToken) 
+{ 
+var replyActivity = MessageFactory.Text("message is soft deleted"); 
+await turnContext.SendActivityAsync(replyActivity, cancellationToken); 
+} 
+
+```
+
+# [JSON](#tab/json5)
+
+```json
+
+{
+"type":"messageDelete",
+"timestamp":"2022-10-28T17:19:43.1612052Z",
+"localTimestamp":"2022-10-28T10:19:43.1612052-07:00",
+"id":"1666977568748",
+"channelId":"msteams",
+"serviceUrl":"https://canary.botapi.skype.com/amer/",
+"from": {
+    "id":"29:1BLjP9j3_TM4mubmQZsYEo7jDyLeLf_YVA9sVPVO7KMAFMjJWB_EUGveb9EVDh9LgoNp9qjnzEBy4kgw83Jf1Kg",
+    "name":"Alex Wilber",
+    "aadObjectId":"976e4d1e-2108-43ee-a092-46a9507c5606"
+},
+"conversation":{
+    "conversationType":"personal",
+    "tenantId":"528dbe3f-15e0-4e37-84a1-00cc305847dd","id":"a:1tewuGJ44RkB90tiJNQ_I4q8vyuN5CYA_f-v6f0Vd-Bs3Ce85C73Ah1y8TvyjESsTHWjjgw-gnsuIuCUOWkfOCq6qaUYsk2_-fj93XXXHUMAUzhFFvTnaCU7V4WiMqXQL"
+},
+"recipient":{
+    "id":"28:0d469698-ab9d-479a-b0d8-758b6e6b1235",
+    "name":"Testbot"
+},
+"entities":[
+    {
+        "locale":"en-US",
+        "country":"US",
+        "platform":"Web",
+        "timezone":"America/Los_Angeles",
+        "type":"clientInfo"
+    }
+],
+"channelData":{
+    "eventType":"softDeleteMessage",
+    "tenant":{"id":"528dbe3f-15e0-4e37-84a1-00cc305847dd"}
+},
+"locale":"en-US",
+"localTimezone":"America/Los_Angeles"
+}  
+
+```
+
+# [Javascript](#tab/javascript5)
+
+You can use one of the following methods to get event notifications to handle the message updates using the Bot SDK:
+
+**​Event function registration**:
+
+```javascript
+
+this.onTeamsMessageSoftDeleteEvent(async (context, next) => {
+    let deletedMessage = context.activity.text; // undefined
+    let messageId = context.activity.id;
+    let previousMessage = this.currentMessage[messageId];
+
+      await context.sendActivity(`The deleted message was "${previousMessage}"`);
+
+      this.updateCurrentMessage(messageId, deletedMessage);
+
+    next();
+})
+
+```
+
+**​Method override**:
+
+```javascript
+
+async onTeamsMessageSoftDelete(context) {
+    let deletedMessage = context.activity.text; // undefined
+    let messageId = context.activity.id;
+    let previousMessage = this.currentMessage[messageId];
+
+    await context.sendActivity(`The deleted message was "${previousMessage}"`);
+
+    this.updateCurrentMessage(messageId, deletedMessage);
```

Comment:
`this.currentMessage` and `this.updateCurrentMessage` don't make much sense without the entire class definition or any context. It would be best to just remove it for this case. The changes to make should be 

Delete lines 519, 521 and 525. And update the string on line 523 to `"The deleted message id is ${messageId}"`

Reply:
The same applies to the other method of implementation as well

Reply:
updated as suggested.